data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/validator/GetAggregate.java

```diff
@@ -0,0 +1,146 @@
+
+  public GetAggregate(final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
+    this.jsonProvider = jsonProvider;
+    this.provider = provider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Aggregates all attestations matching given attestation data root and slot.",
+      tags = {TAG_VALIDATOR},
+      queryParams = {
+        @OpenApiParam(
+            name = ATTESTATION_DATA_ROOT,
+            description =
+                "`String` HashTreeRoot of AttestationData that validator want's aggregated.",
+            required = true),
+        @OpenApiParam(
+            name = SLOT,
+            description = "`UnsignedLong` Non-finalized slot for which to create the aggregation.",
+            required = true)
+      },
+      description = "Aggregates all attestations matching given attestation data root and slot.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = Attestation.class, isArray = true),
+            description =
+                "Returns aggregated `Attestation` object with same `AttestationData` root."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Invalid parameter supplied"),
+        @OpenApiResponse(
+            status = RES_FORBIDDEN,
+            description = "Beacon node was not assigned to aggregate on that subnet"),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR, description = "Beacon node internal error.")
+      })
+  @Override
+  public void handle(Context ctx) throws Exception {
+
+    try {
+      final Map<String, List<String>> parameters = ctx.queryParamMap();
+      if (parameters.size() < 2) {
+        throw new IllegalArgumentException(
+            String.format("Please specify both %s and %s", ATTESTATION_DATA_ROOT, SLOT));
+      }
+      Bytes32 beacon_block_root = getParameterValueAsBytes32(parameters, ATTESTATION_DATA_ROOT);
+      @SuppressWarnings("unused")
+      UnsignedLong slot = getParameterValueAsUnsignedLong(parameters, SLOT);
+
+      ctx.result(
+          provider
+              .createAggregate(beacon_block_root)
+              .thenApplyChecked(optionalAttestation -> serializeResult(ctx, optionalAttestation))
+              .exceptionallyCompose(error -> handleError(ctx, error)));
+    } catch (final IllegalArgumentException e) {
+      ctx.result(jsonProvider.objectToJSON(new BadRequest(e.getMessage())));
+      ctx.status(SC_BAD_REQUEST);
+    }
+  }
+
+  private String serializeResult(final Context ctx, final Optional<Attestation> optionalAttestation)
+      throws com.fasterxml.jackson.core.JsonProcessingException {
+    if (optionalAttestation.isPresent()) {
+      ctx.status(SC_OK);
+      String json = jsonProvider.objectToJSON(optionalAttestation.get());
+      return json;
+    } else {
+      ctx.status(SC_NOT_FOUND);
+      return "";
+    }
+  }
+
+  /*
+   At the moment we aren't handling the error:
```

Comment:
```suggestion
   At the moment we aren't enforcing:
```
Just so it doesn't sound like we might get an error and fail to handle it.